### DISCUSSION TITLE
Problems restoring after Postgres update 13>14

### DISCUSSION THREAD

**Discussion by w0lf3h** [teslamate-org/teslamate#2784] [Q&A]
Ugh. I was under the impression I was doing all this correctly, but I'm still getting the "no data" in all my Grafana tabs after restore. Been trying to look in the logs too, and I don't seem to be getting any errors? (Or even any indication that restore was even trying to occur.)

So I have my teslamate.bck file from the dump from postgres 13.
The command I used initially:
`docker exec teslamate_database_1 pg_dump -U teslamate teslamate > teslamate.bck`

I was having trouble finding the volume to remove, so I `docker volume ls` to rm the database volume. 
`docker volume rm teslamate_teslamate-db`

Changed the database version and then re-upped the container. So now I have working Teslamate containers but essentially a completely fresh instance (no data).

***

So then I stop teslamate containers,

Use this code to reinitialize:
```
docker-compose exec -T teslamate_database_1 psql -U teslamate << .
drop schema public cascade;
create schema public;
create extension cube;
create extension earthdistance;
CREATE OR REPLACE FUNCTION public.ll_to_earth(float8, float8)
    RETURNS public.earth
    LANGUAGE SQL
    IMMUTABLE STRICT
    PARALLEL SAFE
    AS 'SELECT public.cube(public.cube(public.cube(public.earth()*cos(radians(\$1))*cos(radians(\$2))),public.earth()*cos(radians(\$1))*sin(radians(\$2))),public.earth()*sin(radians(\$1)))::public.earth';
.
```

Then restore:
`docker-compose exec -T teslamate_database_1 psql -U teslamate -d teslamate < teslamate.bck`

And restart!

****

Does it look like I'm doing something wrong? Because for some reason, I'm just getting no data panels in Grafana, and all Teslamate data is gone as well (geofences, etc.)
Do I need to enter a database password somewhere in the restoration process for it to take? I don't understand what I'm doing wrong!

**Comment by CrowMash** [comment]
Have you checked your backup file? Does it look ok?

> So then I stop teslamate container**s**,

Did you stop all containers or just the teslamate one? The database container must be up.

> (Or even any indication that restore was even trying to occur.)

You should have some sort of logging when you execute SQL request. If you don't, maybe your database container is not up.

**Reply by w0lf3h** [reply]
Visually (to me) it looks OK. Starts with `PostgreSQL database dump` and ends with `PostgreSQL database dump complete` :woman_shrugging:

I did actually try it both ways. Makes sense that the database container has to be up though. 

I was actually just going thru my logs in Portainer for my teslamate_database_1 container. The container is up, but logs don't necessarily display anything after running the `docker-compose exec -T teslamate_database_1 psql -U teslamate -d teslamate < teslamate.bck` command.

Just got home from work and tried it from the beginning. `docker-compose down` all containers, remove database volume, then `docker-compose up`, stop teslamate_teslamate_1 container, then attempting restoring and restarting the teslamate container. Nada. Not sure what I'm doing wrong, but I assume it's something, and I'm not getting any errors.... It's very weird, really.

**Reply by natrlhy** [reply]
If you are upgrading the Postgres version:

https://docs.teslamate.org/docs/maintenance/upgrading_postgres

Don't forget:

docker volume rm "$(basename "$PWD")_teslamate-db"

Follow this closely for backup/restore:

https://docs.teslamate.org/docs/maintenance/backup_restore

**Reply by w0lf3h** [reply]
> If you are upgrading the Postgres version:
> 
> https://docs.teslamate.org/docs/maintenance/upgrading_postgres
> 
> Don't forget:
> 
> docker volume rm "$(basename "$PWD")_teslamate-db"
> 
> Follow this closely for backup/restore:
> 
> https://docs.teslamate.org/docs/maintenance/backup_restore

Is this different than just removing the teslamate_teslamate-db volume? I just used `docker volume ls` and removed it that way.

**Comment by w0lf3h** [comment]
Ugh. I figured it out. My backup file was not inside of the container. :facepalm: :facepalm: Took way too long to figure out.

**Reply by swizzlest** [reply]
I'm having this same issue, but I'm not sure I understand what you figured out.  I'd appreciate if you could add a bit more detail to dumb it down for me.  Thank you in advance!

**Reply by michaelcarvalho86** [reply]
Yes, what's the meaning of: "My backup file was not inside of the container"?

I have the same issue, I restore my backup file using:

[https://docs.teslamate.org/docs/maintenance/backup_restore](https://docs.teslamate.org/docs/maintenance/backup_restore)

and NADA!

I copy my file directly in the folder: /home/pi

My backup was done in a different version and now I reinstalled everything from the beginning... This could be the problem?
